src/CognitiveServices/CognitiveServices.Test/ScenarioTests/CognitiveServicesAccountTests.ps1

```diff
@@ -322,6 +322,219 @@ function Test-NewAzureRmCognitiveServicesAccountWithCustomDomain
     }
 }
 
+<#
+.SYNOPSIS
+Test New-AzCognitiveServicesAccountKey
+#>
+function Test-NewAzureRmCognitiveServicesAccountWithVnet
+{
+    # Setup
+    $rgname = Get-CognitiveServicesManagementTestResourceName;
+
+    try
+    {
+        # Test
+        $accountname = 'csa' + $rgname;
+        $skuname = 'S2';
+        $accounttype = 'TextAnalytics';
+        $loc = Get-Location -providerNamespace "Microsoft.CognitiveServices" -resourceType "accounts" -preferredLocation "West Central US";
+
+        New-AzResourceGroup -Name $rgname -Location $loc;
+
+		$networkRuleSet = [Microsoft.Azure.Commands.Management.CognitiveServices.Models.PSNetworkRuleSet]::New()
+		$networkRuleSet.AddIpRule("200.0.0.0")
+		$networkRuleSet.AddVirtualNetworkRule("/subscriptions/f9b96b36-1f5e-4021-8959-51527e26e6d3/resourceGroups/control-plane-unit-test/providers/Microsoft.Network/virtualNetworks/vnet-unit-test/subnets/default")
```

Comment:
Hi @cormacpayne 
I tried to import Network.csproj, but it seems to have a lot of upgrades in many projects and some failed to upgrade.

I check Storage project, they actually didn't cover these Cmdlets in UT. I'm wondering maybe we can either use hard-coded resources temporary or remove this tests.

Reply:
@yangyuan I've seen this upgrade issue before in VS when trying to add a reference to another project, so I've been manually adding projects to solutions and the SDK to the test project.

As an example, DataLake references the Network project in their solution [here](https://www.fun-center.com/tukwila/attractions/) and [here](https://github.com/Azure/azure-powershell/blob/master/src/DataLakeStore/DataLakeStore.sln#L41-L44), so these additions would also need to be made in your solution file. You will also need to reference the SDK in your test project so you can instantiate a mock Network client from their SDK, which can be seen [here](https://github.com/Azure/azure-powershell/blob/master/src/DataLakeStore/DataLakeStore.Test/DataLakeStore.Test.csproj#L15), [here](https://github.com/Azure/azure-powershell/blob/master/src/DataLakeStore/DataLakeStore.Test/ScenarioTests/AdlsTestsBase.cs#L106) and [here](https://github.com/Azure/azure-powershell/blob/master/src/DataLakeStore/DataLakeStore.Test/ScenarioTests/AdlsTestsBase.cs#L157-L163).

Even though the Storage tests use hard-coded values, we should be able to avoid this by the changes suggested above and make it so the tests are easier to record for all users moving forward.

Reply:
Did what you suggested and it works.
Thanks for the suggestion.